### PR TITLE
fix static check failures in staging  pkg

### PR DESCRIPTION
**Pull request by carlory**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

```
Errors from staticcheck:

vendor/k8s.io/cloud-provider/service/helpers/helper_test.go:62:3: this value of cidrs is never used (SA4006)

vendor/k8s.io/sample-controller/controller_test.go:179:2: unreachable case clause: k8s.io/kubernetes/vendor/k8s.io/client-go/testing.CreateAction will always match before k8s.io/kubernetes/vendor/k8s.io/client-go/testing.UpdateAction (SA4020)

vendor/k8s.io/cli-runtime/pkg/resource/builder_test.go:59:2: var metaAccessor is unused (U1000)
vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:47:2: only the first constant in this group has an explicit type (SA9004)
```
**Which issue(s) this PR fixes**:

Ref: #81657

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
@liggitt is the `omitemtpy` concerning enough to get this into 1.16?

**Comment by liggitt** [comment]
Already fixed in https://github.com/kubernetes/kubernetes/pull/81359/files, right?

**Comment by carlory** [comment]
@dims @liggitt 

Thanks for your review. I rebased codes and pushed again.

**Comment by dims** [comment]
```
/home/prow/go/src/k8s.io/kubernetes/hack/.staticcheck_failures is not in alphabetical order. Please sort it:

  LC_ALL=C sort -o /home/prow/go/src/k8s.io/kubernetes/hack/.staticcheck_failures /home/prow/go/src/k8s.io/kubernetes/hack/.staticcheck_failures

```

**Comment by carlory** [comment]
@dims thanks for your guide

File: staging/src/k8s.io/sample-controller/controller_test.go [hunk 1]
```diff
@@ -180,15 +180,6 @@ func checkAction(expected, actual core.Action, t *testing.T) {
 		expObject := e.GetObject()
 		object := a.GetObject()
 
-		if !reflect.DeepEqual(expObject, object) {
-			t.Errorf("Action %s %s has wrong object\nDiff:\n %s",
-				a.GetVerb(), a.GetResource().Resource, diff.ObjectGoPrintSideBySide(expObject, object))
-		}
-	case core.UpdateAction:
```
Comment (cheftako) [inline staging/src/k8s.io/sample-controller/controller_test.go]: Doesn't this mean we are no longer checking expectUpdateDeploymentAction()? 

Related this switch has no default case with fail which means it is easy for us to ignore things we are not explicitly coding for.

Comment (carlory) [reply staging/src/k8s.io/sample-controller/controller_test.go]: thanks for your review. It's my mistake.  ` expectUpdateDeploymentAction()` should be checked. 

we should use `XXXActionImpl` instead `Interface`.

```go
	switch a := actual.(type) {
	case core.CreateActionImpl:
		e, _ := expected.(core.CreateActionImpl)
		expObject := e.GetObject()
		object := a.GetObject()

		if !reflect.DeepEqual(expObject, object) {
			t.Errorf("Action %s %s has wrong object\nDiff:\n %s",
				a.GetVerb(), a.GetResource().Resource, diff.ObjectGoPrintSideBySide(expObject, object))
		}
	case core.UpdateActionImpl:
		e, _ := expected.(core.CreateActionImpl)
		expObject := e.GetObject()
		object := a.GetObject()

		if !reflect.DeepEqual(expObject, object) {
			t.Errorf("Action %s %s has wrong object\nDiff:\n %s",
				a.GetVerb(), a.GetResource().Resource, diff.ObjectGoPrintSideBySide(expObject, object))
		}
	case core.PatchActionImpl:
		e, _ := expected.(core.PatchActionImpl)
		expPatch := e.GetPatch()
		patch := a.GetPatch()

		if !reflect.DeepEqual(expPatch, patch) {
			t.Errorf("Action %s %s has wrong patch\nDiff:\n %s",
				a.GetVerb(), a.GetResource().Resource, diff.ObjectGoPrintSideBySide(expPatch, patch))
		}
	}
```

File: staging/src/k8s.io/cli-runtime/pkg/resource/builder_test.go [hunk 2]
```diff
@@ -49,14 +49,13 @@ import (
 	utiltesting "k8s.io/client-go/util/testing"
 
 	// TODO we need to remove this linkage and create our own scheme
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (hwdef) [inline staging/src/k8s.io/cli-runtime/pkg/resource/builder_test.go]: This also changed unrelated line.
Please keep the PR scope simple and clean for the PR message and the corresponding issue.

Comment (carlory) [reply staging/src/k8s.io/cli-runtime/pkg/resource/builder_test.go]: removed
